### ISSUE TITLE
Fido2 Conformance Test Tool: android-safetynet certificate contains invalid AuthorityKeyIdentifier

### ISSUE THREAD

**Issue by xepa** [fido-alliance/conformance-tools-issues#438]
By submitting this issue you are acknowledging that any information regarding this issue will be publicly available.

If you have privacy concerns, please email [conformance-tools@fidoalliance.org](mailto:conformance-tools@fidoalliance.org)

## What protocol and version of the protocol are you testing?

UAF/U2F/FIDO2 v1.0/v1.1/v1.2
FIDO2 android-safetynet
## What is your implementation class?

Server/Client/ASM+Authr
Server
## What is the version of the tool are you using?
v0.10.207 (latest for Mac)

## What is the OS and the version are you running?
Mac 10.14.3

## Issue description
the AuthorityKeyIdentifier part of the 1st certificate sent in the x5c header of the jwt seems to have an incorrect syntax according to https://www.ietf.org/rfc/rfc5280.txt (4.2.1.1) 

Expected:
```
  AuthorityKeyIdentifier ::= SEQUENCE {
      keyIdentifier             [0] KeyIdentifier           OPTIONAL,
      authorityCertIssuer       [1] GeneralNames            OPTIONAL,
      authorityCertSerialNumber [2] CertificateSerialNumber OPTIONAL  }
KeyIdentifier ::= OCTET STRING
```

```
see hex encoded: 30168014d2c4b0d291d44c1171b361cb3da1fedda86ad4e3
https://lapo.it/asn1js/#MBaAFNLEsNKR1EwRcbNhyz2h_t2oatTj
```
Actual:
```
AuthorityKeyIdentifier ::=  OCTET STRING
```
```
see hex encoded: 0416a014263f2e62d4126ddb9f8aeb19e674ed84b4f85ab2
https://lapo.it/asn1js/#BBagFCY_LmLUEm3bn4rrGeZ07YS0-Fqy
```
for reference full certificate:
```
-----BEGIN CERTIFICATE-----
MIIFPDCCBCSgAwIBAgIPBCGxAcQuQWN2PsG/19VmMA0GCSqGSIb3DQEBCwUAMG4x
CzAJBgNVBAYTAlVTMSswKQYDVQQKDCJGSURPIEFsbGlhbmNlcyBGQUtFIFRydXN0
IFNlcnZpY2VzMTIwMAYDVQQDDClGSURPIEFsbGlhbmNlcyBGQUtFIEludGVybmV0
IEF1dGhvcml0eSBGMTAeFw0xODAyMDEwMDAwMDBaFw0yMDAyMjIwNzQzMTZaMHEx
CzAJBgNVBAYTAlVTMQswCQYDVQQIDAJNWTESMBAGA1UEBwwJV2FrZWZpZWxkMRYw
FAYDVQQKDA1GSURPIEFsbGlhbmNlMQwwCgYDVQQLDANDV0cxGzAZBgNVBAMMEmF0
dGVzdC5hbmRyb2lkLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEB
AOVv1nC7bHQjVR2Flg0GZtqePD4J+x1vqwFh6FpOUe5sBsZ1a0K4Vme7Y6QScaxM
Y018rbfaXKMnaVYUeO5ouMWvoZJHMVVsh9y29X6a2MTog+sXEVJ1QcxWsCv1QLkc
CJdPYYQlj8t3MPS9XAcR25nq8mDGPEgOlIhrvI4mVPtB4s+xJOoVQ4Hz/i1Po9tW
WL2YCIEEar/OI+UkM5TwhT5lCmNZr2tpVoCj0ksWOfiId9PU/MrLO9uATXbyfncs
jLToJzRb13HbCO+7eim6XqH8Et7qxbAF4t8rwHBpeSbR/xci7OMuPSJinQLFEzRs
Anaod0i66lopGaxW7lcDZM8CAwEAAaOCAdIwggHOMCEGA1UdIAQaMBgwDAYKKwYB
BAHWeQIFAzAIBgZngQwBAgIwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDAYDVR0TAQH/
BAIwADAfBgNVHQ4EGAQWwhS5WZbKA6QhWYHwr06PV7fuaB+H5jAfBgNVHSMEGAQW
oBQmPy5i1BJt25+K6xnmdO2EtPhasjBTBgNVHR8ETDBKMEigRqBEhkJodHRwczov
L2ZpZG9hbGxpYW5jZS5jby5uei9zYWZldHluZXRwa2kvY3JsL2F0dGVzdC5hbmRy
b2lkLmNvbS5jcmwwge4GCCsGAQUFBwEBBIHhMIHeMGsGCCsGAQUFBzAChl9odHRw
czovL2ZpZG9hbGxpYW5jZS5jby5uei9zYWZldHluZXRwa2kvRklETyUyMEZha2Ul
MjBSb290JTIwQ2VydGlmaWNhdGUlMjBBdXRob3JpdHklMjAyMDE4LmNydDBvBggr
BgEFBQcwAYZjaHR0cHM6Ly9maWRvYWxsaWFuY2UuY28ubnovc2FmZXR5bmV0cGtp
L2NybC9GSURPJTIwRmFrZSUyMFJvb3QlMjBDZXJ0aWZpY2F0ZSUyMEF1dGhvcml0
eSUyMDIwMTguY3JsMA0GCSqGSIb3DQEBCwUAA4IBAQCQ/q5Rvgz/EikzaXvNhVrV
2dkiQVPOY1lbqW0fVZdiXNJD/8jbB2klfmadO1bUsvJeHKl0a1utLj4U8eOuBCy8
aasEuojUmpI4tYV98YX1SsBl7xz/ox7w9tCOz6FEgNEaZydCfKPjkBH4JA5jv/Tj
nolnBFyyhPyRVmEbIQcuwmAXnrTeiCCiQ7/Zyao/cW2+/C3eZEiBcRAgMD1x4xfW
Kj3j18WcneuwRwsmtVkHiyOI6JBUrWxi0g7cc2PK6a6ycbE4AVbgo3TcqvQ6dWkK
sFfsjp0yXiOYdMoz2197ooObLskyki9tvlDFLy6imqlGddUGPJWspFD7EHkOGvbZ
-----END CERTIFICATE-----
```

**Comment by aseigler** [comment]
Running into this same issue as well.  Demonstration: https://play.golang.org/p/FjB31Akv8WG

**Comment by herrjemand** [comment]
Correction @xepa @aseigler. It 's 1.0.0

**Comment by aseigler** [comment]
This still appears broken in 1.0.0.

**Comment by xepa** [comment]
I ran 0.10.212 (the version that I reported this with) .. and it reported that it worked around one of these certificates both in androidsafetynet.  Running the new 1.0.0 version only reports that it worked around one of these certificates.

I would assume that there is another certificate there that has not been fixed.
I can supply the certificate if this speeds up the process.

**Comment by aseigler** [comment]
The attestation cert appears to be ok now, it is the intermediate cert ("C=US, O=FIDO Alliances FAKE Trust Services, CN=FIDO Alliances FAKE Internet Authority F1") that is still broken.  This one:

-----BEGIN CERTIFICATE-----
MIIFDzCCA/egAwIBAgIPBOp8pEPNwcw8feZIEM+CMA0GCSqGSIb3DQEBCwUAMFAx
CzAJBgNVBAYTAlVTMRYwFAYDVQQKDA1GSURPIEFsbGlhbmNlMSkwJwYDVQQDDCBG
SURPIEFsbGlhbmNlcyBGQUtFIFJvb3QgQ0EgLSBTMTAeFw0xNzAyMDEwMDAwMDBa
Fw0zNTAxMzEyMzU5NTlaMG4xCzAJBgNVBAYTAlVTMSswKQYDVQQKDCJGSURPIEFs
bGlhbmNlcyBGQUtFIFRydXN0IFNlcnZpY2VzMTIwMAYDVQQDDClGSURPIEFsbGlh
bmNlcyBGQUtFIEludGVybmV0IEF1dGhvcml0eSBGMTCCASIwDQYJKoZIhvcNAQEB
BQADggEPADCCAQoCggEBAJfUvjcWSta/6CdbFwXlTrAeiG0BDjoHzDn5Yx7R6Djq
eLHyCxD5VYxeav/rPqPAebpcncFp0CJZyfoT32uIYk1TNtxpvXm4j9HizBbSU6YL
As2Dz7WHC6BSqDh2JDq4cor7TDp6AC1S/DAgNp4R5Xn4JuGO7tuhwyBTWWzcGPr3
VvLIPf9pyAcmjR5hjw4v1E8MuYROSWgWEAYY+pjnR1XEueGbuL8WZ5W2NtFQlQYv
bNTDscYJTcNCyt3OWf9MtNFzbj0tSteQp+pa+biI/+anHSBHjn6XInwmOWohPx1j
ubcXVAGjqSUtjZCmTGOGRdNZUkVigwIFcldzpTR+S3UCAwEAAaOCAcYwggHCMAsG
A1UdDwQEAwIBhjBLBgNVHSAERDBCMEAGBmeBDAECAjA2MDQGCCsGAQUFBwIBFiho
dHRwczovL2ZpZG9hbGxpYW5jZS5jby5uei9zYWZldHluZXRwa2kvMB0GA1UdJQQW
MBQGCCsGAQUFBwMBBggrBgEFBQcDAjASBgNVHRMBAf8ECDAGAQH/AgEAMB8GA1Ud
DgQYBBbCFCY/LmLUEm3bn4rrGeZ07YS0+FqyMB8GA1UdIwQYBBagFKo2/0Fp0p2K
O9gcZdNb3mtwLZSQMHQGA1UdHwRtMGswaaBnoGWGY2h0dHBzOi8vZmlkb2FsbGlh
bmNlLmNvLm56L3NhZmV0eW5ldHBraS9jcmwvRklETyUyMEZha2UlMjBSb290JTIw
Q2VydGlmaWNhdGUlMjBBdXRob3JpdHklMjAyMDE4LmNybDB7BggrBgEFBQcBAQRv
MG0wawYIKwYBBQUHMAGGX2h0dHBzOi8vZmlkb2FsbGlhbmNlLmNvLm56L3NhZmV0
eW5ldHBraS9GSURPJTIwRmFrZSUyMFJvb3QlMjBDZXJ0aWZpY2F0ZSUyMEF1dGhv
cml0eSUyMDIwMTguY3J0MA0GCSqGSIb3DQEBCwUAA4IBAQAvH6ElB/udPgVqiabr
mBciHKaB9R6Pe4hIXK7fMJktp0XN0UYQHKdxf8m4c0HqFrMsTOgN+uijcMHBBDTC
i+oZwxCeVfUz8a9dQBtlODrJkcNVqIXgLxBbfkN/BXelxRfMji/DFLUhYxtJHHVv
ZrlJWYwaI3f7SsSzhHEqG/rJqjvVzPHmxjSgSRl1Ll405fY85J053f37Tjf4Tzx3
YmS5BBZEQYjx9FSD03qDAkP2EQAoK9DxdbF1iHi/1kZ8LHSAZj3SXy7lkQNyWqjK
benGC+rF7s2W1AQ4j47BXwR1R1ocEDV7ZJ2vuKoyH/kD/0Oij6TJ7br8sXDaBf4O
hk+I
-----END CERTIFICATE-----

**Comment by herrjemand** [comment]
New build in the morning. Sorry for the delay guys

**Comment by aseigler** [comment]
Have not seen new build yet.

**Comment by herrjemand** [comment]
Fixed in 1.1.1
